### ISSUE TITLE
Support dynamic setting of DropBlock.dropout_rate

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#176]

**Comment by LukeWood** [comment]
Currently the constant value is baked into the graph, I suggest a schedule parameter to support this.

**Comment by kksinghal** [comment]
I want to look into this.

**Comment by sebastian-sz** [comment]
@LukeWood What could this API look like? Do we want to support only linear schedule, mentioned in the paper?
`layer = DropBlock2D(schedule={"total_steps": 100, "starting_rate": 0.0}`

Or maybe accept a separate class that will handle scheduling in it's own call method?
```python
class DropoutRateScheduler:
    def __init__(self, total_steps: int, starting_rate = 0.0):
        self._current_step = 0

    def __call__(current_rate):
          self._current_step += 1
          # modify `current_rate` accordingly


layer = DropBlock2D(scheduler=DropoutRateScheduler(...))
```

Or maybe something completely different?

**Comment by LukeWood** [comment]
I think we want to also support something like lambda functions:
```
DropBlock2D(schedule=lambda current_step: current_step / 1000)
```

or something

**Comment by kksinghal** [comment]
@sebastian-sz, I don't think it's worth making a separate DropoutRateScheduler class since it is a very generic function, it returns a linear function of a value. Otherwise, we could also use any scheduler function, like tf.keras.optimizers.schedules.PolynomialDecay, with power=1.

Instead,
1. The user can pass the initial dropout rate, end dropout rate, total train steps as arguments, and we can handle the linear scheduler inside our DropBlock2D class so that user don't have to care about the scheduler. 
2. The user can also pass a custom-defined scheduler function, which takes the current step as input and returns the rate, as @LukeWood mentioned. Here, the user can also use the tf schedulers functions.
So, the user can go either way. What do you say?

**Comment by kksinghal** [comment]
@LukeWood, Regarding #184, 
Do you want me to add arguments of linear scheduler to DropBlock2D? or do you want to pass the scheduler function only.
About #184, Do you want to create a new class of schedulers like tf.keras.optimizers.schedules?
Please mention if you are working on it. Otherwise, I would like to work on both of these.

**Comment by LukeWood** [comment]
Lets create a schedule that works like tf.keras.optimizers.schedules.

Feel free to create a class in the PR and I will provide API feedback in the code review.  Thank you @kksinghal.

So...

DropBlock will take a `Schedule` class optionally as a parameter.  Another option would be to offer a keras Callback similar to https://github.com/keras-team/keras/issues/16234

@fchollet thoughts on using the callback that recompiles opposed to supporting a first party schedule class? I think the schedule class is probably cleaner for the end user.

**Comment by LukeWood** [comment]
FYI @kksinghal ^

**Comment by kksinghal** [comment]
Actually, tensorflow schedulers are exactly what we want, so we could these scheduler functions, but the problem is they are accessed by tf.keras.optimizers.schedules, and they are named as learning_rate, and it isn't right to use them, so maybe we can use [this](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/optimizer_v2/learning_rate_schedule.py), and make the necessary changes for our purpose. Which schedulers do you want me to implement? In the paper, they use only linear.

**Comment by LukeWood** [comment]
The KerasCV team is attempting to narrow the package scope in order to more efficiently use our limited resources. As such, we are only accepting contributions directly related to items on the 6 month roadmap for the package.

If you feel this feature could be particularly useful, or would like to contribute it - please feel free to re-open this issue and ping LukeWood, ianstenbit or tanzhenyu

Feel free to read more on the roadmap:
https://github.com/keras-team/keras-cv/blob/master/.github/ROADMAP.md
